helix-view/src/document.rs

```diff
@@ -1080,22 +1080,25 @@ impl Document {
     }
 
     pub fn pickup_last_saved_time(&mut self) {
-        self.last_saved_time = match self.path.as_mut().unwrap().metadata() {
-            Ok(metadata) => match metadata.modified() {
-                Ok(mtime) => mtime,
-                Err(_) => {
+        self.last_saved_time = match self.path().as_mut() {
```

Comment:
It's unrelated to these changes but the `as_mut` is unnecessary since `Path::metadata` takes a `&Path`

```suggestion
        self.last_saved_time = match self.path() {
```

Reply:
I agree with both the comments, but those apply to the code as it was before the PR. If you still want me to apply the changes I'd be happy to do so.

Reply:
Yeah although they're not totally related to the fix I think it's fine to land them in this PR